targets/TARGET_STM/rtc_api.c

```diff
@@ -247,40 +276,49 @@ void rtc_write(time_t t)
     timeStruct.StoreOperation = RTC_STOREOPERATION_RESET;
 #endif /* TARGET_STM32F1 */
 
+#if DEVICE_LPTICKER && !MBED_CONF_TARGET_LPTICKER_LPTIM
+    /* Need to update LP_continuous_time value before new RTC time */
+    uint32_t current_lp_time = rtc_read_lp();
+
+    /* LP_last_RTC_time value is updated with the new RTC time */
+    LP_last_RTC_time = timeStruct.Seconds + timeStruct.Minutes * 60 + timeStruct.Hours * 60 * 60;
+
+    /* Save current SSR */
+    uint32_t Read_SubSeconds = (uint32_t)(RTC->SSR);
+#endif /* DEVICE_LPTICKER && !MBED_CONF_TARGET_LPTICKER_LPTIM */
+
     // Change the RTC current date/time
     if (HAL_RTC_SetDate(&RtcHandle, &dateStruct, RTC_FORMAT_BIN) != HAL_OK) {
         error("HAL_RTC_SetDate error\n");
     }
     if (HAL_RTC_SetTime(&RtcHandle, &timeStruct, RTC_FORMAT_BIN) != HAL_OK) {
         error("HAL_RTC_SetTime error\n");
     }
+
+#if DEVICE_LPTICKER && !MBED_CONF_TARGET_LPTICKER_LPTIM
+    while (Read_SubSeconds != (RTC->SSR)) {
```

Comment:
Is it possible that SSR increments and you get stuck in this loop?

Reply:
SSR sub seconds register is reset to PREDIV_S value every second.
This prescaler value is set only during init, so I would say we can't get stuck...